### PR TITLE
use terminology *data reader creator* instead of *data reader*, and r…

### PR DESCRIPTION
**Pull request by helinwang**
…emove lambda usage from design doc

### CODE REVIEW

File: doc/design/reader/README.md [hunk 1]
```diff
@@ -54,38 +54,40 @@ Since reading data may take time and training can not proceed without data. It i
...
 ```python
-def data_reader_fake_image():
+def data_reader_creator_fake_image():
 	while True:
 		yield numpy.random.uniform(-1, 1, size=20*20)
 
-def data_reader_bool(t):
-	while True:
-		yield t
+def data_reader_creator_bool(t):
```
Comment (wangkuiyi) [inline doc/design/reader/README.md]: data_reader_bool ==> reader_creator_bool

File: doc/design/reader/README.md [hunk 2]
```diff
@@ -54,38 +54,40 @@ Since reading data may take time and training can not proceed without data. It i
...
+	def creator:
```
Comment (wangkuiyi) [inline doc/design/reader/README.md]: creator ==> reader

File: doc/design/reader/README.md [hunk 3]
```diff
@@ -54,38 +54,40 @@ Since reading data may take time and training can not proceed without data. It i
...
-### Compose Multiple Data Readers
+### Compose Multiple Data Reader Creators
 
 For example, we want to use a source of real images (reusing mnist dataset), and a source of fake images as input for [Generative Adversarial Networks](https://arxiv.org/abs/1406.2661).
 
 We can do:
 
 ```python
-def data_reader_fake_image():
+def data_reader_creator_fake_image():
```
Comment (wangkuiyi) [inline doc/design/reader/README.md]: data_reader_creator_fake_image ==> reader_creator_random_image

File: doc/design/reader/README.md [hunk 4]
```diff
@@ -54,38 +54,40 @@ Since reading data may take time and training can not proceed without data. It i
...
 	while True:
```
Comment (wangkuiyi) [inline doc/design/reader/README.md]: reader_creator_random_image 应该有参数表示图像大小

```
def reader_creator_random_image(width, height):
    def reader():
        while True:
            yield numpy.random.uniform(-1, 1, size=width*height)
```

File: doc/design/reader/README.md [hunk 5]
```diff
@@ -1,35 +1,35 @@
-# Python Data Reader Design Doc
+# Python Data Reading Design Doc
 
-Paddle reads data from data reader during training. It will be passed into `paddle.train` as a parameter.
+Paddle reads data from *data reader* during training. *data reader creator* (or *reader creator*) creates a *data reader* when invoked. *reader creator* will be passed into `paddle.train` as a parameter.
```
Comment (wangkuiyi) [inline doc/design/reader/README.md]: At training and testing time, PaddlePaddle programs need to read data. To ease the users' work to write data readign code, we define that

- A *reader* is a function that reads data (from file, network, random number generator, etc) and yields data items.
- A *reader creator* is a function that returns a reader function.
- A *reader decorator* is a function, which accepts one or more readers, and returns a reader.

and provide frequently used reader creators and reader decorators.

File: doc/design/reader/README.md [hunk 6]
```diff
@@ -1,35 +1,35 @@
...
 
-## Data Reader Interface
+## Data Reader Creator Interface
 
-Data reader is a function with no parameter that creates a iterable (anything can be used in `for x in iterable`):
+Data reader creator is a function with no parameter that creates a iterable (anything can be used in `for x in iterable`):
```
Comment (wangkuiyi) [inline doc/design/reader/README.md]: I think a reader creator can have parameters. For example:

```python
def reader_creator_text(filename):
    def reader():
        ... read from filename ...
    return reader
```

Here the creator does have one paraemter.

File: doc/design/reader/README.md [hunk 7]
```diff
@@ -41,11 +41,11 @@ label_layer = paddle.layer.data("label", ...)
 paddle.train(paddle.dataset.mnist, {"image":0, "label":1}, 128, 10, ...)
 ```
 
-## Data Reader Decorators
+## Data Reader Creator Decorator
```
Comment (wangkuiyi) [inline doc/design/reader/README.md]: Reader Creator Decorator ==> Reader Decorator

File: doc/design/reader/README.md (hunk 2 above)
Comment (helinwang) [inline doc/design/reader/README.md]: Done.

File: doc/design/reader/README.md (hunk 7 above)
Comment (helinwang) [inline doc/design/reader/README.md]: Done.

File: doc/design/reader/README.md (hunk 6 above)
Comment (helinwang) [inline doc/design/reader/README.md]: Done.

File: doc/design/reader/README.md (hunk 5 above)
Comment (helinwang) [inline doc/design/reader/README.md]: Done.

File: doc/design/reader/README.md (hunk 1 above)
Comment (helinwang) [inline doc/design/reader/README.md]: Done.

File: doc/design/reader/README.md (hunk 4 above)
Comment (helinwang) [inline doc/design/reader/README.md]: Done.

File: doc/design/reader/README.md (hunk 3 above)
Comment (helinwang) [inline doc/design/reader/README.md]: Done.

**Pull request merged**